contracts/MarketCollateralPool.sol

```diff
@@ -65,26 +73,54 @@ contract MarketCollateralPool is Ownable {
     /// and issue them the requested qty of long and short tokens
     /// @param marketContractAddress            address of the market contract to redeem tokens for
     /// @param qtyToMint                      quantity of long / short tokens to mint.
+    /// @param isAttemptToPayInMKT            if possible, attempt to pay fee's in MKT rather than collateral tokens
     function mintPositionTokens(
         address marketContractAddress,
-        uint qtyToMint
-    ) external onlyWhiteListedAddress(marketContractAddress) {
+        uint qtyToMint,
+        bool isAttemptToPayInMKT
+    ) external onlyWhiteListedAddress(marketContractAddress)
+    {
 
         MarketContract marketContract = MarketContract(marketContractAddress);
         require(!marketContract.isSettled(), "Contract is already settled");
 
-        uint neededCollateral = MathLib.multiply(qtyToMint, marketContract.COLLATERAL_PER_UNIT());
-        uint feeAmount = MathLib.multiply(qtyToMint, marketContract.FEE_PER_UNIT());
-        uint totalTransferAmount = neededCollateral.add(feeAmount);
         address collateralTokenAddress = marketContract.COLLATERAL_TOKEN_ADDRESS();
+        uint neededCollateral = MathLib.multiply(qtyToMint, marketContract.COLLATERAL_PER_UNIT());
+        // the user has selected to pay fees in MKT and those fees are non zero (allowed) OR
+        // the user has selected not to pay fees in MKT, BUT the collateral token fees are disabled (0) AND the
+        // MKT fees are enabled (non zero).  (If both are zero, no fee exists)
+        bool isPayFeesInMKT = (isAttemptToPayInMKT &&
```

Comment:
@pelsasser Why not simplify and just let the minter specify that he would like to pay in MKT or CollateralToken?

Because it can get confusing when I call mint with `isAttemptToPayInMKT == false`, but because the contract only has mkt_token_fees specified, we still charge his MKT tokens.

~I feel it would be better if we explicitly let the caller specify which token to pay fees and prevent minting if the contract doesn't~

So, I just thought this through now and I realized what you were doing :), this logic ensures that at least one of the non-zero fees specified is used no matter what token the caller specifies. 👌

Reply:
So basically we should ensure that on the user interface (or market.js) we perform some pre-checks and ensure the minter is aware of which fees would actually be used.

Reply:
yea, the minting flow will have very clear information re: the fees and how they can be paid.